### PR TITLE
[css-properties-values-api] Fix syntax string parsing to not treat whitespace between components the same as pipes.

### LINKED ISSUE #893: [css-properties-values-api] The fact that "foo bar" and "foo | bar" are equivalent syntax is confusing.
In https://drafts.css-houdini.org/css-properties-values-api-1/#consume-syntax-descriptor, `|` and any other character are treated basically equivalently, so it means that `<length> <percentage>` is really a syntax for `<length> | <percentage>`, rather than a `<length>`, then a `<percentage>`.

Is this intentional? Is there any reason we're closing the gate to more expressive syntaxes that allow multiple keywords?

### PR DESCRIPTION
**Pull request by emilio**
Fixes #893

### CODE REVIEW

**Comment by andruud** [comment]
Oh, nice, thanks @emilio <3

I'm not actually an editor though, so I can't merge it.

--> @tabatkins 

(It would be nice to have assign-review-to-people privileges, so I didn't have to ping people in a comment, like a commoner, djeez).

File: css-properties-values-api/Overview.bs [hunk 1]
```diff
@@ -596,19 +596,15 @@ Parsing the syntax string {#parsing-syntax}
...
 		:: If |descriptor|’s [=list/size=] is greater than zero,
-			<a>consume a syntax component</a> from |stream|.
-			If failure was returned,
+			and the <a>current input code point</a> is not U+007C VERTICAL LINE (|),
 			return failure;
-			otherwise,
-			[=list/append=] the returned value to |descriptor|.
 
-			If |descriptor|’s [=list/size=] is zero, return failure.
+		:: If the <a>current input code point</a> is not U+007C VERTICAL LINE (|),
```
Comment (tabatkins) [inline css-properties-values-api/Overview.bs]: These should all be paragraphs under the one dd, not multiple separate dds

Comment (emilio) [reply css-properties-values-api/Overview.bs]: Sure, done.

**Comment by emilio** [comment]
I don't seem to have merge access to this repo, so if you could do that @tabatkins it'd be awesome.

Thanks!

**Comment by andruud** [comment]
@emilio Just discovered that this allows a leading "|", e.g. "|foo" is a valid syntax.

**Comment by andruud** [comment]
Let's rethink step 5 as a whole. Drop the "repeatedly consume" thing and instead do:

Step 5:
 * Consume a syntax component from stream. If failure was returned, return failure; otherwise, append the returned value to descriptor. [Note: Consume a syntax component consumes leading whitespace, which is relevant for repeats of this step].
 * Consume as much whitespace as possible from stream.
 * Consume the next input code point from stream:
   * EOF: Return descriptor. [Note: I don't think we still need the descriptor size check, since we'll now exit early if "Consume a syntax component" fails?]
   * U+007C VERTICAL LINE (|):  Repeat this step.
   * Anything else: Return failure.

WDYT?

**Comment by emilio** [comment]
Err, yes indeed it does.

Or alternatively, also reconsume the current code-point even if it's `|`, if the output is empty. But yeah, that sounds cleaner.

**Comment by emilio** [comment]
Your algorithm also needs to skip trailing whitespace after parsing a component, otherwise you stop parsing `foo | bar` unless it's written `foo|bar`

**Comment by emilio** [comment]
Err, you did that already, should learn to read. Yeah, that looks good.

**Pull request merged**